### PR TITLE
66 audit log

### PR DESCRIPTION
**Pull request by Eugene7011**
Updated PR
GitGuardian has uncovered 2 secrets - keys are from Front

### CODE REVIEW

File: goods-partner-backend/pom.xml [hunk 1]
```diff
@@ -88,6 +88,12 @@
             <version>2.17.1</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-text</artifactId>
+            <version>1.9</version>
+        </dependency>
```
Comment (belosh59) [inline goods-partner-backend/pom.xml:95]: Why we need it?

File: goods-partner-backend/src/main/java/com/goodspartner/entity/DeliveryHistoryTemplate.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.goodspartner.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+@AllArgsConstructor
+public enum DeliveryHistoryTemplate {
+    DELIVERY_CREATED("{role} {userEmail} створив(ла) доставку"),
+    DELIVERY_UPDATED("{role} {userEmail} оновив(ла) доставку"),
+    DELIVERY_CALCULATED("{role} {userEmail} розрахував(ла) доставку"),
```
Comment (BogdanParashchak) [inline goods-partner-backend/src/main/java/com/goodspartner/entity/DeliveryHistoryTemplate.java:10]: use the style as below pls

File: goods-partner-backend/src/main/java/com/goodspartner/entity/DeliveryHistoryTemplate.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    ADDRESS_CREATE("${role} ${userEmail} додав(ла) нову адресу ${clientAddress} клієнт ${clientName}"),
+    MAPPOINT_CHANGE("@{role} ${userEmail} змінив(ла) координати в адресі ${clientAddress} клієнт ${clientName} на широта ${latitude}, довгота ${longitude}");
+
+    @Getter
+    private final String template;
+
+    @Override
+    public String toString() {
+        return template;
+    }
```
Comment (BogdanParashchak) [inline goods-partner-backend/src/main/java/com/goodspartner/entity/DeliveryHistoryTemplate.java:25]: do we need this?

File: goods-partner-backend/src/main/java/com/goodspartner/event/DeliveryAuditEventListener.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import com.goodspartner.service.impl.DefaultDeliveryHistoryService;
+import com.goodspartner.util.AuditorBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+
+@Component
+public class DeliveryAuditEventListener {
```
[the remark this answers is not in the document]
Comment (BogdanParashchak) [reply goods-partner-backend/src/main/java/com/goodspartner/event/DeliveryAuditEventListener.java:13]: @belosh59 what do think?

File: goods-partner-backend/src/main/java/com/goodspartner/util/AuditorBuilder.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+        if(authentication != null){
+            if(authentication.getPrincipal().getClass().equals(CustomOAuth2User.class)){
+                String role = ((CustomOAuth2User) authentication.getPrincipal()).getAuthorities().toString();
+                String userEmail = ((CustomOAuth2User) authentication.getPrincipal()).getEmail();
+                values.put("role", role);
+                values.put("userEmail", userEmail);
+            }
+            values.put("role", authentication.getAuthorities().toString());
+            values.put("userEmail", authentication.getPrincipal().toString());
```
Comment (BogdanParashchak) [inline goods-partner-backend/src/main/java/com/goodspartner/util/AuditorBuilder.java:25]: maybe else here?

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultDeliveryHistoryService.java [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    private final OrderExternalMapper orderExternalMapper;
+    private final DeliveryRepository deliveryRepository;
+
+    @Override
+    @Transactional
+    public DeliveryHistoryDto add(DeliveryHistoryDto deliveryHistoryDto) {
+        DeliveryHistory addedDeliveryHistory = deliveryHistoryRepository
+                .save(deliveryHistoryMapper.toDeliveryHistory(deliveryHistoryDto));
+
+        return deliveryHistoryMapper.toDeliveryHistoryDto(addedDeliveryHistory);
```
[the remark this answers is not in the document]
Comment (BogdanParashchak) [reply goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultDeliveryHistoryService.java:52]: @belosh59 what do you think?

**Review by BogdanParashchak — COMMENTED** [review]
see comments

File: goods-partner-backend/src/main/java/com/goodspartner/entity/AddressExternal.java [hunk 7]
```diff
@@ -6,6 +6,7 @@
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
+import lombok.ToString;
```
[the remark this answers is not in the document]
Comment (Eugene7011) [reply goods-partner-backend/src/main/java/com/goodspartner/entity/AddressExternal.java:9]: you are right

File: goods-partner-backend/src/main/java/com/goodspartner/entity/Delivery.java [hunk 8]
```diff
@@ -52,6 +52,9 @@ public class Delivery {
     @OneToMany(mappedBy = "delivery", cascade = CascadeType.ALL, orphanRemoval = true)
     private List<CarLoad> carLoads = new ArrayList<>();
 
+    @OneToMany(mappedBy = "delivery", cascade = CascadeType.ALL)
```
[the remark this answers is not in the document]
Comment (Eugene7011) [reply goods-partner-backend/src/main/java/com/goodspartner/entity/Delivery.java:55]: history can't be without delivery (relation OneToMany). So CascadeType.All will ensure that all events such as persist, refresh, remove that occur on the parent will be passed to the child.

File: goods-partner-backend/src/main/java/com/goodspartner/util/AuditorBuilder.java (hunk 5 above)
Comment (Eugene7011) [reply goods-partner-backend/src/main/java/com/goodspartner/util/AuditorBuilder.java:25]: +++

File: goods-partner-backend/src/main/java/com/goodspartner/entity/DeliveryHistoryTemplate.java (hunk 3 above)
Comment (Eugene7011) [reply goods-partner-backend/src/main/java/com/goodspartner/entity/DeliveryHistoryTemplate.java:25]: yes. I need string template. It doesn't work without toString

File: goods-partner-backend/src/main/java/com/goodspartner/mapper/DeliveryHistoryMapper.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+@Mapper(componentModel = "spring")
+public interface DeliveryHistoryMapper {
+
+    @Mapping(target = "id", source = "id")
+    @Mapping(target = "delivery.id", source = "deliveryId")
+    @Mapping(target = "createdAt", source = "createdAt")
+    @Mapping(target = "role", source = "role")
+    @Mapping(target = "userEmail", source = "userEmail")
+    @Mapping(target = "action", source = "action")
```
[the remark this answers is not in the document]
Comment (Eugene7011) [reply goods-partner-backend/src/main/java/com/goodspartner/mapper/DeliveryHistoryMapper.java]: corrected

File: goods-partner-backend/src/main/java/com/goodspartner/entity/DeliveryHistoryTemplate.java (hunk 3 above)
Comment (Eugene7011) [reply goods-partner-backend/src/main/java/com/goodspartner/entity/DeliveryHistoryTemplate.java]: +

File: goods-partner-backend/src/main/java/com/goodspartner/mapper/DeliveryHistoryMapper.java [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    DeliveryHistory toDeliveryHistory(DeliveryHistoryDto deliveryHistoryDto);
+
+    @Mapping(target = "id", source = "id")
+    @Mapping(target = "deliveryId", source = "delivery.id")
+    @Mapping(target = "createdAt", source = "createdAt")
+    @Mapping(target = "role", source = "role")
+    @Mapping(target = "userEmail", source = "userEmail")
+    @Mapping(target = "action", source = "action")
```
[the remark this answers is not in the document]
Comment (Eugene7011) [reply goods-partner-backend/src/main/java/com/goodspartner/mapper/DeliveryHistoryMapper.java]: deleted not needed

File: goods-partner-backend/src/main/java/com/goodspartner/event/DeliveryAuditEventListener.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (belosh59) [reply goods-partner-backend/src/main/java/com/goodspartner/event/DeliveryAuditEventListener.java:13]: Listener is fine I think

File: goods-partner-backend/src/main/java/com/goodspartner/event/DeliveryAuditEventListener.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Autowired
+    private DefaultDeliveryHistoryService deliveryHistoryService;
```
[the remark this answers is not in the document]
Comment (belosh59) [reply goods-partner-backend/src/main/java/com/goodspartner/event/DeliveryAuditEventListener.java:16]: Interface should be here, also use the @AllArhsConstructor and final variable to inject it

File: goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultDeliveryHistoryService.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (belosh59) [reply goods-partner-backend/src/main/java/com/goodspartner/service/impl/DefaultDeliveryHistoryService.java:45]: No we dont need mapping and return type at all.
The only place it is used is:
```
@EventListener(DeliveryAuditEvent.class)
    public void handleEvent(DeliveryAuditEvent event){
            deliveryHistoryService.add(createNewDeliveryHistoryDto(event));
    }
```

File: goods-partner-backend/src/main/java/com/goodspartner/util/AuditorBuilder.java [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+import com.goodspartner.service.dto.CustomOAuth2User;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class AuditorBuilder {
+
+    public static Map<String, String> getCurrentAuditor() {
```
[the remark this answers is not in the document]
Comment (Eugene7011) [reply goods-partner-backend/src/main/java/com/goodspartner/util/AuditorBuilder.java:12]: ok

File: goods-partner-backend/src/main/java/com/goodspartner/event/DeliveryAuditEventListener.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (Eugene7011) [reply goods-partner-backend/src/main/java/com/goodspartner/event/DeliveryAuditEventListener.java:16]: done

**Review by BogdanParashchak — APPROVED** [review]

**Review by belosh59 — APPROVED** [review]

**Pull request merged**